src/screens/ReceiveCoinsScreen.tsx

```diff
@@ -0,0 +1,188 @@
+            } else {
+              BlueApp.saveToDisk(); // caching whatever getAddressAsync() generated internally
+            }
+            this.setState({
+              address: address,
+              addressText: address,
+            });
+          } else if (wallet.chain === Chain.OFFCHAIN) {
+            try {
+              await Promise.race([wallet.getAddressForTransaction(), BlueApp.sleep(1000)]);
+              address = wallet.getAddress();
+            } catch (_) {}
+            if (!address) {
+              // either sleep expired or getAddressAsync threw an exception
+              console.warn('either sleep expired or getAddressAsync threw an exception');
+              address = wallet.getAddress();
+            } else {
+              BlueApp.saveToDisk(); // caching whatever getAddressAsync() generated internally
+            }
+          }
+          this.setState({
+            address: address,
+            addressText: address,
+          });
+        } else if (wallet.getAddress) {
+          this.setState({
+            address: wallet.getAddress(),
+            addressText: wallet.getAddress(),
+          });
+        }
+      }
+    }
+
+    InteractionManager.runAfterInteractions(async () => {
+      const bip21encoded = bip21.encode(this.state.address);
+      this.setState({ bip21encoded });
+    });
+  }
+
+  updateAmount = (amount: number) => {
+    const bip21encoded = bip21.encode(this.state.address, {
+      amount,
+    });
+    this.setState({
+      amount,
+      bip21encoded,
+    });
+  };
+
+  editAmount = () => {
+    this.props.navigation.navigate(Route.EditText, {
+      title: i18n.receive.header,
+      label: i18n.receive.details.amount,
+      onSave: this.updateAmount,
+      value: this.state.amount,
+    });
+  };
+
+  share = async () => {
+    if (this.qrCodeSVG === undefined) {
+      Share.open({
+        message: `bitcoin:${this.state.address}`,
+      }).catch(error => console.log(error));
+    } else {
+      InteractionManager.runAfterInteractions(async () => {
+        this.qrCodeSVG.toDataURL((data: any) => {
+          const shareImageBase64 = {
+            message: `bitcoin:${this.state.address}`,
+            url: `data:image/png;base64,${data}`,
+          };
+          Share.open(shareImageBase64).catch(error => console.log(error));
+        });
+      });
+    }
+  };
+
+  render() {
+    const { amount, addressText, bip21encoded } = this.state;
+    return (
+      <ScreenTemplate
+        footer={
+          <Button title={i18n.receive.details.share} onPress={this.share} containerStyle={styles.buttonContainer} />
+        }
+      >
+        <View style={styles.qrcontainer}>
+          <QRCode
+            value={bip21encoded}
+            size={130}
+            color={BlueApp.settings.foregroundColor}
+            logoBackgroundColor={BlueApp.settings.brandingColor}
+            ecl={'H'}
+            // @ts-ignore
```

Comment:
I've updated the QRCode library in my PR (merged) and now this `ts-ignore` is not needed anymore :)